recipes/libarchive/all/conanfile.py

```diff
@@ -116,110 +116,118 @@ def requirements(self):
-        self._cmake.definitions["ENABLE_LibGCC"] = False
-        self._cmake.definitions["ENABLE_CNG"] = self.options.with_cng
+            tc.variables["POSIX_REGEX_LIB"] = "LIBPCREPOSIX"
+        tc.variables["ENABLE_LibGCC"] = False
+        tc.variables["ENABLE_CNG"] = self.options.with_cng
         # turn off features
-        self._cmake.definitions["ENABLE_ACL"] = self.options.with_acl
+        tc.variables["ENABLE_ACL"] = self.options.with_acl
         # turn off components
-        self._cmake.definitions["ENABLE_TAR"] = False
-        self._cmake.definitions["ENABLE_CPIO"] = False
-        self._cmake.definitions["ENABLE_CAT"] = False
-        self._cmake.definitions["ENABLE_TEST"] = False
+        tc.variables["ENABLE_TAR"] = False
+        tc.variables["ENABLE_CPIO"] = False
+        tc.variables["ENABLE_CAT"] = False
+        tc.variables["ENABLE_TEST"] = False
         # too strict check
-        self._cmake.definitions["ENABLE_WERROR"] = False
-        if tools.Version(self.version) >= "3.4.2":
-            self._cmake.definitions["ENABLE_MBEDTLS"] = self.options.with_mbedtls
-        self._cmake.definitions["ENABLE_XATTR"] = self.options.with_xattr
+        tc.variables["ENABLE_WERROR"] = False
+        if Version(self.version) >= "3.4.2":
+            tc.variables["ENABLE_MBEDTLS"] = self.options.with_mbedtls
+        tc.variables["ENABLE_XATTR"] = self.options.with_xattr
+
+        if self.settings.os == "Windows" and self.settings.get_safe("os.subsystem") != "cygwin":
+            tc.cache_variables["ZLIB_WINAPI"] = "yes"
 
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        tc.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
+        cmakelists_path = os.path.join(self.source_folder, "CMakeLists.txt")
 
-        cmakelists_path = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        replace_in_file(self, cmakelists_path,
+                        "LIST(APPEND ADDITIONAL_LIBS ${ZLIB_LIBRARIES})",
+                        "LIST(APPEND ADDITIONAL_LIBS ZLIB::ZLIB)")
 
         # it can possibly override CMAKE_MODULE_PATH provided by generator
-        tools.replace_in_file(cmakelists_path,
-                              "SET(CMAKE_MODULE_PATH",
-                              "LIST(APPEND CMAKE_MODULE_PATH")
+        replace_in_file(self, cmakelists_path,
+                        "SET(CMAKE_MODULE_PATH",
+                        "LIST(APPEND CMAKE_MODULE_PATH")
         # allow openssl on macOS
         if self.options.with_openssl:
-            tools.replace_in_file(cmakelists_path,
-                                  "IF(ENABLE_OPENSSL AND NOT CMAKE_SYSTEM_NAME MATCHES \"Darwin\")",
-                                  "IF(ENABLE_OPENSSL)")
+            replace_in_file(self, cmakelists_path,
```

Comment:
I feel like these could be patches bit of have no clue if that's reasonable

Reply:
I can look into that once I get the pipeline passing

Reply:
That was painful, but done.